libtiledbsoma/src/soma/soma_array.h

```diff
@@ -1587,6 +1567,9 @@ class SOMAArray : public SOMAObject {
     // Array associated with mq_
     std::shared_ptr<Array> arr_;
 
+    // Array schema
```

Comment:
Does this open up a possibility for `arr_` and `schema_` to be out of sync?

Sorry & maybe I missed something but when would we ever want `schema_` to be anything other than `arr_->schema()`? And if the answer is "never", then can we simply refer to `arr_->schema()` rather than `schema_`?

Reply:
I forgot to comment about this. If the array is not open or opened in write mode, then we can't access `arr_->schema()` directly.

Reply:
You're right though that this could open up a possibility for the schema in `arr_` and `schema_` to be out of sync since we utilize schema evolutions. I could remove this `schema_` member and then modify the `schema()` accessor to use `arr_->schema()` if already open for writes and then open the array to grab the schema as necessary otherwise?

Reply:
Ooh nice! ✨

Reply:
But @nguyenv  if you're saying the answer to my question is 'we need a copy of the schema that was obtained when the array was opened for read, even when the array is currently opened for write and so the schema is otherwise inaccessible" -- then that's a great answer, and we can keep separate `arr_` and `schema_` as you have now. Just please add this information as a code comment in `managed_query.h` saying this.